internal/scanners/vmss/rules.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package vmss
+
+import (
+	"strings"
+
+	"github.com/Azure/azqr/internal/scanners"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+)
+
+// GetRules - Returns the rules for the VirtualMachineScaleSetScanner
+func (a *VirtualMachineScaleSetScanner) GetRules() map[string]scanners.AzureRule {
+	return map[string]scanners.AzureRule{
+		"vmss-001": {
```

Comment:
VMs Scale Sets and VM's with diagnostic extensions enabled do not create `/providers/microsoft.insights/diagnosticSettings` child resources and therefore this check will not work. Let's remove it for this first version.

Reply:
Good to know, removed it!